src/dataEntities/errors.ts

```diff
@@ -21,6 +21,36 @@ export class UnreachableCaseError extends ApplicationError {
     }
 }
 
+/**
+ * Base class for errors that ar thrown when a deeper error cannot be handled.
+ * Customizes the stack trace in order to show the full stack trace of both the current
+ * error and the originating error.
+ */
+export class NestedError extends ApplicationError {
```

Comment:
Yeah, that seems a good idea, I will try merging NestedError into ApplicationError

Reply:
Done in b71fcd5f6b00041be7989b3b7bf43253c9f7978a.
It is slightly awkward to have to pass `undefined` as middle parameter in all subclasses, but I wanted to preserve the possibility of using `ApplicationError` directly for generic errors, like
```
   throw new ApplicationError("Something bad happened");

   throw new ApplicationError("Something bad made something else bad happen", doh);
```
So the `name` parameter needs to be the last.

Reply:
Note that in most subclasses I am actually disabling the nestedError parameter (as it often does not make sense), but it could easily be added back, should the need arise.